### ISSUE TITLE
Simple typescript pulumi project attempting to create policy.PolicyPack hangs indefinitely 

### ISSUE THREAD

**Issue by alfred-stokespace** [pulumi/pulumi#10381]
### What happened?

A very simple new aws-typescript (or plain typescript) project attempting to use something as simple as 

```
new policy.PolicyPack("aws-tags-policies", {
    enforcementLevel: 'advisory',
    policies: []

});
```
will hang upon `pulumi up` 
```
     Type                 Name          Plan        Info
     pulumi:pulumi:Stack  tagsalot-aws  running..   38531

```

### Steps to reproduce

1. `pulumi new`
2. `aws-typescript` or `typescript` (both tested)
3. name, desc, stack name... doesn't matter
4. aws region (doesn't matter, for just typescript skip this)
5. add any version to package.json `"version": "0.0.1",` 
6. add dependency `npm install @pulumi/policy`
7. replace basic index.js with ... 
    ```
    import * as policy from "@pulumi/policy"

    new policy.PolicyPack("aws-tags-policies", {
        enforcementLevel: 'advisory',
        policies: []
    });
    ```
8. pulumi up
9. wait for ... minutes.... hours? 

### Expected Behavior

create the PolicyPack quickly without error

### Actual Behavior

indefinite hang 

```

pulumi/common/test$ pulumi up
Previewing update (dev)

View Live: https://app.pulumi.com/REDACTED/test/dev/previews/REDACTED

     Type                 Name      Plan       Info
 +   pulumi:pulumi:Stack  test-dev  create..   33751

```

This just goes on for ever.

If you use an already existing project you get indefinite hang on `running`... instead of `create...`

### Versions used

v3.37.2

```
{
    "name": "test",
    "version": "0.0.1",
    "main": "index.ts",
    "devDependencies": {
        "@types/node": "^14"
    },
    "dependencies": {
        "@pulumi/aws": "^5.0.0",
        "@pulumi/awsx": "^0.40.0",
        "@pulumi/policy": "^1.4.0",
        "@pulumi/pulumi": "^3.0.0"
    }
}

```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by RobbieMcKinstry** [comment]
Hi @alfred-stokespace Thanks for filing. I'm able to reproduce the bug. We'll look and see why this happens for empty policy lists.

**Comment by alfred-stokespace** [comment]
Thanks @RobbieMcKinstry I wanted to add that it's not just empty policy that repro's. 
I first came across this behavior with a populated `policies` array like so, just tested and is behaving identical (ie. indefinite hang)

```
import * as policy from "@pulumi/policy";
import {isTaggable} from "./taggable";

new policy.PolicyPack("aws-tags-policies", {
    enforcementLevel: 'advisory',
    policies: [
        {
            name: "check-required-tags",
            description: "Ensure required tags are present on all taggable resources.",
            configSchema: {
                properties: {
                    requiredTags: {
                        type: "array",
                        items: { type: "string" },
                    },
                },
            },
            validateResource: (args, reportViolation) => {
                const config = args.getConfig<AwsTagsPolicyConfig>();
                const requiredTags = config.requiredTags;
                if (requiredTags && isTaggable(args.type)) {
                    const ts = args.props["tags"];
                    for (const rt of requiredTags) {
                        if (!ts || !ts[rt]) {
                            reportViolation(`Taggable resource '${args.urn}' is missing required tag '${rt}'`);
                        }
                    }
                }
            },
        }
    ]

});

interface AwsTagsPolicyConfig {
    requiredTags?: string[];
}
```
most of that is just copy/pasted from https://github.com/joeduffy/aws-tags-example/tree/4658142a68b19510bd1b1af8d15819e449302eea/autotag-ts

**Comment by Frassle** [comment]
Hey sorry for the confusion here, but you can't currently use `@pulumi/policy` inline in a Pulumi program. Policies are currently expected to be separate from the program itself in a "policy pack" folder. See https://www.pulumi.com/docs/guides/crossguard/ for the documentation about how to work with policy.

**Comment by alfred-stokespace** [comment]
I initially was following the blog post by @joeduffy https://www.pulumi.com/blog/automatically-enforcing-aws-resource-tagging-policies/
I didn't notice any mention in that blog post about CrossGaurd, so that's confusing if you intend people to understand CrossGaurd first.  
I assumed I could succeed by following the coding examples mentioned in that blog post https://github.com/joeduffy/aws-tags-example/tree/master/policy-pack-ts

I do see now that when I try to run the policy pack command that I'm required to have a PulumiPolicy.yaml file which the code samples do have. 

Am I on the right track?  I'm assuming this will lead me to a functional tag policy that I can assign to other stacks.

**Comment by alfred-stokespace** [comment]
okay.. so I need a pulumi project, otherwise I get...
```
error: no Pulumi.yaml project file found (searching upwards from ). If you have not created a project yet, use `pulumi new` to do so
```
However the coding examples don't have that? https://github.com/joeduffy/aws-tags-example/tree/master/policy-pack-ts
```
So I'm in a directory with a pulumi project...

```
alfreds@DESKTOP-A4HMCBO:/REDACTED/pulumi/common/tagsalot$ pulumi up --policy-pack=./ --policy-pack-config=./basic-policy.json
Previewing update (REDACTED/aws)

View Live: https://app.pulumi.com/REDACTED/tagsalot/aws/previews/bae9f899-6724-42de-8b16-e9382185eb88

     Type                 Name          Plan        Info
     pulumi:pulumi:Stack  tagsalot-aws  running..   34595
```
and I'm back to a hang... 

So I'm still not sure how to succeed here. I guess I abandon the blog post and go read through the CrossGaurd docs and hopefully that will spell out in detail what to do?

**Comment by alfred-stokespace** [comment]
Okay I think I understand now... the `getting started` on cross-guard makes sense. 
Basically, the confusing thing that isn't mentioned in the blog and it's code examples is the barrier around `Pulumi Business Critical` and the dynamic around local policy evaluation. 
So I was able to do both local policy evaluation of the policy project (w/policy pack) against another project which had resources relevant to the policy project's policy pack and I was able to publish that policy pack to Pulumi service. 

lots of P's
